agent/api/ecsclient/client.go

```diff
@@ -494,7 +507,39 @@ func (client *APIECSClient) buildContainerStateChangePayload(change api.Containe
 		statechange.ExitCode = aws.Int64(exitCode)
 	}
 
+	networkBindings := getNetworkBindings(change, shouldExcludeIPv6PortBinding)
+	// we enforce a limit on the no. of network bindings for containers with at-least 1 port range requested.
+	// this limit is enforced by ECS, and we fail early and don't call SubmitContainerStateChange.
+	if change.Container.GetContainerHasPortRange() && len(networkBindings) > ecsMaxNetworkBindingsLength {
```

Comment:
But this limit technically is effective even there's no port range? If cx specified 100 standalone container ports, there still will be 100 port mappings, and thus the limit should apply?

Reply:
the limit is based on a worst case char limit given that customer uses a port range. In case of singular ports, the limit would be higher since the worst case no. of chars used for a single port are much less than that for a range.

example: `containerPort`: 65535 vs `containerPortRange`: `65534-65535`

We do not enforce a limit for singular ports today and with this feature, we don't want to change behavior for the same.

Reply:
Hmm okay... so this is assuming there are 100 portBindings, all have port ranges in them. I suppose I'd like if we can do something like
```
maxCharSingularPortBinding=A
maxCharRangePortBinding=B
maxCharPortBindingCombined=C  // Control Plane limit
...
if (numOfSingularPortBindings * maxCharSingularPortBinding +numOfRangePortBindings * maxCharRangePortBinding > maxCharPortBindingCombined) 
    // don't send STSC
```
This helps maximize allowed cases, and fail fast whenever possible
But I suppose it's hard to quantify the limit for customer in that case (and how they may avoid hitting it)

Reply:
> Hmm okay... so this is assuming there are 100 portBindings, all have port ranges in them. I suppose I'd like if we can do something like

this limit assumes that there is atleast 1 range. based on feedback from ECS backend team, we don't want to expose specific ECS char limits and instead have user-friendly limit on network bindings defined.